pkg/config/configobserver/config_secret.go

```diff
@@ -0,0 +1,142 @@
+package configobserver
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/config"
+)
+
+// ConfigSecret defines the configuration loaded from cluster secret
+type ConfigSecret struct {
```

Comment:
`ConfigSecret` name seems misleading to me TBH. I know we read the config from the secret (even though the secret is not required and I guess it's not present in most cases) but I think I would omit the word `secret` here.